operator/cmd/main.go

```diff
@@ -127,6 +143,16 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Create a new ProgramHandler to handle Program objects.
+	pHandler := newProgramHandler(mgr.GetClient(), programFSAdvAddr)
+	// Start a simple file-server to serve CR files as compressed tarballs.
+	go func() {
+		// Wait until our manager is elected leader before starting the file server.
+		<-mgr.Elected()
+
+		startProgramFileServer(pHandler, programFSAddr)
+	}()
```

Comment:
nit: I think you can add arbitrary runnables to the manager via its `Add` method, and it will start/stop the runnable like it does the controllers...assumedly after leader election.